version: '3'

services:
  db:
    image: postgres:14-alpine3.17
    container_name: db_app
    env_file:
      - .env
    volumes:
      - ./data/db:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready']
      interval: 10s
      timeout: 5s
      retries: 5

  redis:
    image: redis:7.0.2-alpine
    container_name: redis_app
    command: redis-server --save 20 1 --loglevel warning
    ports:
      - "6379:6379"
    volumes:
      - ./data/cache:/data

  app:
    build:
      context: .
      dockerfile: ./docker/Dockerfile
    network_mode: host
    image: app
    container_name: app_container
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_started
    env_file:
      - .env
    ports:
      - "8000:8000"
    volumes:
      - .:/config
    command: python manage.py runserver 0.0.0.0:8000
